messaging/src/test/java/org/axonframework/eventhandling/GapAwareTrackingTokenTest.java

```diff
@@ -291,7 +403,270 @@ void noMemoryLeakWhenJumpingLargeGaps() {
         assertEquals(1_234, advancedToken.getGaps().size());
     }
 
+
     private TreeSet<Long> asTreeSet(Long... elements) {
         return new TreeSet<>(asList(elements));
     }
+
+    /**
+     * Tests demonstrating how lowerBound() can be used to detect whether an event
+     * was a gap at some reference point (tokenAtReset).
+     * <p>
+     * Key insight: When computing lowerBound(), if the minimum index falls on a gap,
+     * the algorithm walks backwards to find the first non-gap position. This behavior
+     * naturally distinguishes:
+     * <ul>
+     *   <li>Events that were gaps (lowerBound walks back → different from newToken)</li>
+     *   <li>Events that were already processed (lowerBound stays → same as newToken)</li>
+     * </ul>
+     * <p>
+     * This is used by ReplayToken to determine if an event is truly "new" (gap-fill)
+     * vs a "replay" (already seen before reset).
+     */
+    @Nested
+    class LowerBoundForGapDetection {
```

Comment:
Again, not a fan of having replay details in the gap aware tracking token test class. Perhaps this should be a separate test class, specifically to validate the lower bound behavior of the gap aware tracking token for replay specifics. Or, we remove replay specifics from these tests, of course.